pkg/tfgen/docs.go

```diff
@@ -876,13 +877,100 @@ func cleanupDoc(g *generator, info tfbridge.ResourceOrDataSourceInfo, doc parsed
 }
 
 var markdownLink = regexp.MustCompile(`\[([^\]]*)\]\(([^\)]*)\)`)
-var codeLikeSingleWord = regexp.MustCompile("([\\s`\"\\[])(([0-9a-z]+_)+[0-9a-z]+)([\\s`\"\\]])")
+var codeBlocks = regexp.MustCompile(`(?ms)\x60\x60\x60[^\n]*?$.*?\x60\x60\x60\s*$`)
+var codeLikeSingleWord = regexp.MustCompile(`(([\s"\[])(([0-9a-z]+_)+[0-9a-z]+)([\s"\]]))|(\x60(([0-9a-z]+_)*[0-9a-z]+)\x60)`)
 
 // Regex for catching reference links, e.g. [1]: /docs/providers/aws/d/networ_interface.html
 var markdownPageReferenceLink = regexp.MustCompile(`\[[1-9]+\]: /docs/providers(?:/[a-z1-9_]+)+\.[a-z]+`)
 
 const elidedDocComment = "<elided>"
 
+func fixupPropertyReferences(language language, pkg string, info tfbridge.ProviderInfo, text string) string {
+	fixupText := func(text string) string {
+		return codeLikeSingleWord.ReplaceAllStringFunc(text, func(match string) string {
+			parts := codeLikeSingleWord.FindStringSubmatch(match)
+
+			var open, name, close string
+			if parts[2] != "" {
+				open, name, close = parts[2], parts[3], parts[5]
+			} else {
+				open, name, close = "`", parts[7], "`"
+			}
+			if language == pulumiSchema {
+				if open == "[" || open == "`" {
+					open, close = "", ""
+				}
+			}
+
+			if resInfo, hasResourceInfo := info.Resources[name]; hasResourceInfo {
+				// This is a resource name
+				resname, mod := resourceName(info.GetResourcePrefix(), name, resInfo, false)
+				modname := extractModuleName(mod)
+				switch language {
+				case golang, python:
+					// Use `ec2.Instance` format
+					return open + modname + "." + resname + close
+				case pulumiSchema:
+					// Use a hyperlink to the resource.
+					return fmt.Sprintf("%[1]s[`%[2]s`](#/resources/%[2]s)%[3]s",
+						open, url.PathEscape(string(resInfo.Tok)), close)
+				default:
+					// Use `aws.ec2.Instance` format
+					return open + pkg + "." + modname + "." + resname + close
+				}
+			} else if dataInfo, hasDatasourceInfo := info.DataSources[name]; hasDatasourceInfo {
+				// This is a data source name
+				getname, mod := dataSourceName(info.GetResourcePrefix(), name, dataInfo)
+				modname := extractModuleName(mod)
+				switch language {
+				case golang, python:
+					// Use `ec2.getAmi` format
+					return open + modname + "." + getname + close
+				case pulumiSchema:
+					// Use a hyperlink to the function.
+					return fmt.Sprintf("%[1]s[`%[2]s`](#/functions/%[2]s)%[3]s",
+						open, url.PathEscape(string(dataInfo.Tok)), close)
+				default:
+					// Use `aws.ec2.getAmi` format
+					return open + pkg + "." + modname + "." + getname + close
```

Comment:
I think we'll need to check if `modName` is empty here for package-level references.